packages/cursorless-engine/src/languages/LanguageDefinitions.ts

```diff
@@ -1,31 +1,56 @@
 import {
   Disposable,
-  FileSystem,
   Notifier,
   Range,
   TextDocument,
-  getCursorlessRepoRoot,
   isTesting,
   showError,
+  type IDE,
+  type RawTreeSitterQueryProvider,
+  type Listener,
 } from "@cursorless/common";
-import { join } from "pathe";
+import { toString } from "lodash-es";
 import { SyntaxNode } from "web-tree-sitter";
 import { TreeSitter } from "../typings/TreeSitter";
-import { ide } from "../singletons/ide.singleton";
 import { LanguageDefinition } from "./LanguageDefinition";
-import { toString } from "lodash-es";
 
 /**
  * Sentinel value to indicate that a language doesn't have
  * a new-style query definition file
  */
 const LANGUAGE_UNDEFINED = Symbol("LANGUAGE_UNDEFINED");
 
+export interface LanguageDefinitions {
```

Comment:
I removed the lifecycle methods from this interface (`init` and `dispose`). I don't think interfaces should expose lifecycle methods. Clients of an interface shouldn't be responsible for the lifecycle of any given instance; that is the job of whoever constructs the actual class

Reply:
Very true